STest produces warnings when cross compiling for bare metal

| | |
|:---|:---|
|**_F´ Version_**| e289647d89454cd1fa0e2ee1e06e2873885932b4|
|**_Affected Component_**|  STest  |
---
## Problem Description

I receive the following compiler errors when I cross compile with my bare metal compiler.

```
fprime/STest/STest/Random/Random.cpp:81:44: warning: format '%u' expects argument of type 'unsigned int', but argument 2 has type 'U32' {aka 'long unsigned int'} [-Wformat=]
   81 |         printf("[STest::Random] Read seed %u from file\n", seedValue);
      |                                           ~^               ~~~~~~~~~
      |                                            |               |
      |                                            unsigned int    U32 {aka long unsigned int}
      |                                           %lu
fprime/STest/STest/Random/Random.cpp:85:49: warning: format '%u' expects argument of type 'unsigned int', but argument 2 has type 'U32' {aka 'long unsigned int'} [-Wformat=]
   85 |         printf("[STest::Random] Generated seed %u from system time\n", seedValue);
      |                                                ~^                      ~~~~~~~~~
      |                                                 |                      |
      |                                                 unsigned int           U32 {aka long unsigned int}
      |                                                %lu
fprime/STest/STest/Random/bsd_random.c:287:36: error: format '%ld' expects argument of type 'long int', but argument 3 has type 'size_t' {aka 'unsigned int'} [-Werror=format=]
  287 |       "random: not enough state (%ld bytes); ignored.\n", n);
      |                                  ~~^                      ~
      |                                    |                      |
      |                                    long int               size_t {aka unsigned int}
      |                                  %d
```
## How to Reproduce

1. Include STest into one of your components.
2. Add an STest dependency in your CMakeLists.txt file.
3. Cross compile with bare metal compiler.

## Expected Behavior

I do not expect these warnings to occur.

--- a/STest/STest/Random/Random.cpp
+++ b/STest/STest/Random/Random.cpp
@@ -10,6 +10,7 @@
 // ======================================================================
 
 #include <cassert>
+#include <cinttypes>
 #include <ctime>
 
 #include "STest/Random/Random.hpp"
@@ -37,7 +38,7 @@ namespace STest {
         bool result = true;
         FILE *fp = fopen(fileName, "r");
         if (fp != nullptr) {
-          result = (fscanf(fp, "%u", &value) == 1);
+          result = (fscanf(fp, "%" PRIu32, &value) == 1);
           (void) fclose(fp);
         }
         else {
@@ -59,7 +60,7 @@ namespace STest {
         if (fp != nullptr) {
           int status = fprintf(
               fp,
-              "%u\n",
+              "%" PRIu32 "\n",
               seedValue
           );
           result = (status > 0);
@@ -78,11 +79,11 @@ namespace STest {
       const bool seedValueOK =
         SeedValue::getFromFile("seed", seedValue);
       if (seedValueOK) {
-        printf("[STest::Random] Read seed %u from file\n", seedValue);
+        printf("[STest::Random] Read seed %" PRIu32 " from file\n", seedValue);
       }
       else {
         seedValue = SeedValue::getFromTime();
-        printf("[STest::Random] Generated seed %u from system time\n", seedValue);
+        printf("[STest::Random] Generated seed %" PRIu32 " from system time\n", seedValue);
       }
       (void) SeedValue::appendToFile("seed-history", seedValue);
       SeedValue::set(seedValue);

--- a/STest/STest/Random/bsd_random.c
+++ b/STest/STest/Random/bsd_random.c
@@ -284,7 +284,7 @@ bsd_initstate(seed, arg_state, n)
 		state[-1] = MAX_TYPES * (rptr - state) + rand_type;
 	if (n < BREAK_0) {
 		(void)fprintf(stderr,
-		    "random: not enough state (%ld bytes); ignored.\n", n);
+		    "random: not enough state (%zd bytes); ignored.\n", n);
 		return(0);
 	}
 	if (n < BREAK_1) {

